plugins/prometheus/pkg/data/data.go

```diff
@@ -171,19 +168,18 @@ func GetMetrics(ctx context.Context, dynamicClient dynamic.Interface, restMapper
 	if err != nil {
 		return nil, err
 	}
-	workloadMap := make(map[string]controller.Workload)
+	workloadMap := make(map[string]*controller.Workload)
 	for _, workload := range workloads {
 		for _, pod := range workload.Pods {
-			workloadMap[fmt.Sprintf("%s/%s", pod.GetNamespace(), pod.GetName())] = workload
+			workloadMap[fmt.Sprintf("%s/%s", pod.GetNamespace(), pod.GetName())] = &workload
```

Comment:
@rbren I think this will return same pointer address all the time

Reply:
Maybe we don't need this PR anymore 😆

Reply:
😬 I always mess this up